react/features/speaker-stats/reducer.js

```diff
@@ -0,0 +1,109 @@
+// @flow
+
+import _ from 'lodash';
+
+import { ReducerRegistry } from '../base/redux';
+
+import {
+    INIT_SEARCH,
+    UPDATE_STATS,
+    REORDER_STATS
+} from './actionTypes';
+
+/**
+ * The initial state of the feature speaker-stats.
+ *
+ * @type {Object}
+ */
+const INITIAL_STATE = {
+    stats: {},
+    criteria: ''
+};
+
+ReducerRegistry.register('features/speaker-stats', (state = _getInitialState(), action) => {
+    switch (action.type) {
+    case INIT_SEARCH:
+        return _updateCriteria(state, action);
+    case UPDATE_STATS:
+        return _updateStats(state, action);
+    case REORDER_STATS:
+        return _reorderStats(state, action);
+    }
+
+    return state;
+});
+
+/**
+ * Gets the initial state of the feature speaker-stats.
+ *
+ * @returns {Object}
+ */
+function _getInitialState() {
+    return INITIAL_STATE;
+}
+
+/**
+ * Reduces a specific Redux action INIT_SEARCH of the feature
+ * speaker-stats.
+ *
+ * @param {Object} state - The Redux state of the feature speaker-stats.
+ * @param {Action} action - The Redux action INIT_SEARCH to reduce.
+ * @private
+ * @returns {Object} The new state after the reduction of the specified action.
+ */
+function _updateCriteria(state, { criteria }) {
+    return _.assign(
+        {},
+        state,
+        { criteria },
+    );
+}
+
+/**
+ * Reduces a specific Redux action UPDATE_STATS of the feature
+ * speaker-stats.
+ *
+ * @param {Object} state - The Redux state of the feature speaker-stats.
+ * @param {Action} action - The Redux action UPDATE_STATS to reduce.
+ * @private
+ * @returns {Object} The new state after the reduction of the specified action.
+ */
+function _updateStats(state, { stats }) {
+    const finalStats = _.cloneDeep(state.stats);
```

Comment:
What does this method do? Can you explain in the docs in more details?
Why do we need to do a cloneDeep here, is it necessary? 
Later we do _.clone( on an object that had already been cloned by filterBySearchCriteria, can we avoid and that one?

Reply:
This method is called to update the stats without changing their order, because the speaker stats order is dependent on the object properties order and resets on each update if no reordering function is called. But yes, you are correct, it seems that _.clone is enough, so I removed the _.cloneDeep function.